physical/raft/snapshot.go

```diff
@@ -545,12 +544,8 @@ func loadSnapshot(db *bolt.DB, logger log.Logger, snapshotFile io.ReadCloser, pa
 					return err
 				}
 
-				if pathsToFilter != nil {
-					keyToLookUp := []byte(entry.Key)
-					_, _, ok := pathsToFilter.Root().LongestPrefix(keyToLookUp)
-					if ok {
-						continue
-					}
+				if filterKey != nil && filterKey(entry.Key) {
+					continue
```

Comment:
Are we allowing **any** path to be filtered? I guess it doesn't really make sense to prevent users to filter some system specific paths, right?

Reply:
It's a fair point, but I think in the context of this code, there's no knowledge of what the paths are. The physical/raft stuff just thinks of the paths as arbitrary keys, and it should be up to the caller to define what keys should be kept and what shouldn't be.